app/Models/Foundation/Summit/Events/Presentations/PresentationCategory.php

```diff
@@ -229,6 +232,126 @@ public function setTitle($title)
      */
     protected $selection_lists;
 
+    /**
+     * @ORM\ManyToOne(targetEntity="models\summit\PresentationCategory", inversedBy="children", fetch="EXTRA_LAZY")
+     * @ORM\JoinColumn(name="ParentPresentationCategoryID", referencedColumnName="ID")
+     * @var PresentationCategory
+     */
+    protected $parent;
+
+    /**
+     * @ORM\OneToMany(targetEntity="models\summit\PresentationCategory", mappedBy="parent", cascade={"persist","remove"}, fetch="EXTRA_LAZY")
+     */
+    protected $children;
+
+    /**
+     * @return PresentationCategory|null
+     */
+    public function getParent(): ?PresentationCategory {
+        return $this->parent;
+    }
+
+    /**
+     * @return int
+     */
+    public function getParentId(): int
+    {
+        try{
+            if(is_null($this->parent)) return 0;
+            return $this->parent->getId();
+        }
+        catch(\Exception $ex){
+            return 0;
+        }
+    }
+
+    /**
+     * @param PresentationCategory $parent
+     */
+    public function setParent(PresentationCategory $parent) {
+        $this->parent = $parent;
+    }
+
+    public function clearParent() {
+        $this->parent = null;
+    }
+
+    /**
+     * @return int
+     */
+    private function getChildrenMaxOrder(): int
+    {
+        $criteria = Criteria::create();
+        $criteria->orderBy(['order' => 'DESC']);
+        $child = $this->children->matching($criteria)->first();
+        return $child === false ? 0 : $child->getOrder();
+    }
+
+    /**
+     * @return PresentationCategory[]|ArrayCollection
+     */
+    public function getChildren() {
+        return $this->children;
+    }
+
+    /**
+     * @param int $child_id
+     * @return PresentationCategory|null
+     */
+    public function getChildById(int $child_id): ?PresentationCategory {
+        $criteria = Criteria::create();
+        $criteria->where(Criteria::expr()->eq('id', $child_id));
+        $child = $this->children->matching($criteria)->first();
+        return $child === false ? null : $child;
+    }
+
+    /**
+     * @param PresentationCategory $child
+     * @return $this
+     */
+    public function addChild(PresentationCategory $child): PresentationCategory {
+        if($this->children->contains($child)) return $this;
+        $child->setOrder($this->getChildrenMaxOrder() + 1);
+        $child->setParent($this);
+        $this->children->add($child);
+        return $this;
+    }
+
+    /**
+     * @param PresentationCategory $child
+     * @return $this
+     */
+    public function removeChild(PresentationCategory $child): PresentationCategory {
+        if(!$this->children->contains($child)) return $this;
+        $this->children->removeElement($child);
```

Comment:
you must recalculate the ordering here like
https://github.com/OpenStackweb/summit-api/blob/e5ee249a4eb63b653cff82a273e637a35f1b05be/app/Models/Foundation/Summit/Summit.php#L5573

Reply:
done